### DISCUSSION TITLE
mail fetching trigger

### DISCUSSION THREAD

**Discussion by fasatcc** [creecros/Mailmagik#32] [Q&A]
I'm having the Issue configuring the automatic mail conversion into tasks. 
What I have configured so far: 
- IMAP Server
- Task Mail (functioning)
- automatic Action in project config

When now sending a mail (With "Parse from the "SUBJECT" field" enabled) The Task is not being converted immediately. I've waited 5 minutes, not touching the email, and nothing happened. 
Email Subject was this: [Project#33] Test2 c:SQL p:3 d:2023-02-28 s:2023-02-20

**Comment by creecros** [comment]
Are you sure your cron is functioning? How often is it triggered? Which cron are you using?

**Reply by fasatcc** [reply]
Ok, I think there is my confusion in setting it up. 
I have to set up a cronjob like yours `* * * * * cd /var/www/app && ./cli mailmagik:fetchmail` right?

**Reply by fasatcc** [reply]
I have included this line in crontab. But there is some confusion. Why do I need an MTA installed on my local machine to access mails on another mail server? Can you give me a small how to for my next steps?

**Reply by creecros** [reply]
The cron is a background worker on the server that you set to run at specific times, how often is up to you. KB comes with 1 already, and it is generally setup to run once per day, which is how a lot of actions are run, notifications etc...we added a separate cron with mailmagik, that you can utilize because you may want to run the action to retrieve mails more often than the other cron, that said, in order to use it, you have set it up and make sure your action is using the correct cron you want to use. 

I can't say with certainty exactly what your cron should look like, as your KB server may be different than mine. 

You can test it tho, just plop that into you command line, see if it triggers. 

`cd /var/www/app && ./cli mailmagik:fetchmai`

If it does, then it is correct and should work every minute using * * * * *

**Reply by creecros** [reply]
> I have included this line in crontab. But there is some confusion. Why do I need an MTA installed on my local machine to access mails on another mail server? Can you give me a small how to for my next steps?

I would not call the cronjob an MTA, but like you said, the mail server is on another machine. The cron is what tells Mailmagik when to check that mail, and how often. Without it, it will never check for any mails to convert. The task email feature works without the cron, because you are the one telling Mailmagik to check for mails.

**Reply by fasatcc** [reply]
Ok thanks, this helped. I confused the message of the cron Info with the output of the command `.cli mailmagik:fetchmail`. So when directing the command into the kanboard directory, the job worked fine. It saw the messages and marked them as seen. Now I have to get the conversion right. I will inform about my full though process later. So that others that are as stupid as me have an easier time understanding. :)

**Reply by fasatcc** [reply]
And this is where my luck ends. 
I have set up the conversion in the automatic actions. 
I have set up the cronjob (It marks the emails as seen)
But the conversion isn't happening. Any Log recommendation where I can look into (tried syslog and apache/error.log already). Or any help?

**Reply by creecros** [reply]
Check and make sure that the email you are using to send said mail, is your email for the user  in KB, and has permissions in said project.

**Reply by creecros** [reply]
And if you got any screenshots, it might help. Screenshot everything. I might see something.

**Reply by creecros** [reply]
As far as logs go, all I can say is, if there is an error happening, it will show up in the stderr log.

I dont ever use apache, and I have no idea if you are sending stderr logs anywhere. So, I have no idea how to direct you here.

**Reply by fasatcc** [reply]
Ok, I have checked the email of my account. This is the same mail as in my profile and the one that I used to set up mailmagik.

![Automatic action](https://github.com/creecros/Mailmagik/assets/130677126/3d965424-5c8a-442a-8824-47091d9e20e0)
![cronjob](https://github.com/creecros/Mailmagik/assets/130677126/42b2d2f9-9fd8-4ac9-a17d-4811d16ef9b4)
![mailmagick conf](https://github.com/creecros/Mailmagik/assets/130677126/0255ff6e-c9ad-49c6-8451-ae93ac1da88e)

**Reply by creecros** [reply]
Now show me an email you used as a test.

**Reply by creecros** [reply]
> Ok thanks, this helped. I confused the message of the cron Info with the output of the command `.cli mailmagik:fetchmail`. So when directing the command into the kanboard directory, the job worked fine. It saw the messages and marked them as seen. Now I have to get the conversion right. I will inform about my full though process later. So that others that are as stupid as me have an easier time understanding. :)

Also, let's go back to here. What do you mean by "directing the command into kanboard directory"?

Is this where we are at? It marks as seen, but there is no conversion?

**Reply by fasatcc** [reply]
> > Ok thanks, this helped. I confused the message of the cron Info with the output of the command `.cli mailmagik:fetchmail`. So when directing the command into the kanboard directory, the job worked fine. It saw the messages and marked them as seen. Now I have to get the conversion right. I will inform about my full though process later. So that others that are as stupid as me have an easier time understanding. :)
> 
> Also, let's go back to here. What do you mean by "directing the command into kanboard directory"?

I mean the `var/www/html/kanboard/` portion of the cronjob. I cd'd into the directory where the kanboard is stored and not into `var/www/app`.

**Reply by creecros** [reply]
Ya, for it to mark them as seen, means it has made it through the entire actions code and completed, as that is the last step. So, it's definitely not erroring out. 

Couple questions.
1. Using the Task email feature, can you convert an email to a task?
2. Can you show me your subject line, that you are parsing?
3. Maybe something to try, change the parse to the TO field, and see if you can get that to work, just as a test.

**Reply by creecros** [reply]
The only thing I can come up that would prevent the conversion at this point, assuming my understanding of the situation is correct.

1. You are sending to the wrong project, or the project Id is being read as null
2. The user sending the email to be converted, doesn't have permissions in the project.
3. its is working, but for some reason you arent noticing. This is doubtful, but maybe you are filtering them out somehow in the board, or your cache needs to be cleard. Highly unlikely this is the case, but might as well think about it.
  

That's about it.

Everything you've told me leads me to this conclusion. 

Anything else, would cause a visible error in the stderr log, and most likely cause a break, and the mail would not be marked as SEEN.

To start the conversion and find the mails, means you have the correct parse for "Project#", but does not mean you have the correct actual number of the project. 

To reach the end, and mark as seen, means that the code completed, with no errors that would break the code.

**Reply by fasatcc** [reply]
Ok. This is weird now. I'm testing again and it works. I can't say why, because I don't know. Sometimes it needs time. 
Thanks for the help.
